src/Microsoft.AspNet.OData.Shared/Builder/StructuralTypeConfigurationOfTStructuralType.cs

```diff
@@ -359,6 +359,7 @@ public void HasDynamicProperties(Expression<Func<TStructuralType, IDictionary<st
             _configuration.AddDynamicPropertyDictionary(propertyInfo);
         }
 
+  
```

Comment:
Where's the fluent API to config the Instance Annotation container?

Reply:
There isn't one. If you have a property of type IODataInstanceAnnotationContainer then we use it; if you don't, we don't. There isn't any difference in the model (i.e., no equivalent of an open type) so we don't need anything special in model builder.

Reply:
However, in the Edm model, we use "ODataInstanceAnnotationContainerAnnotation", that's the metadata annotation.
Such metadata annotation will be used in Serialization/Deserialization, if we have metadata annotation, we can get "IODataInstanceAnnotationContainer".

If we don't have the fluent API, how to create the "ODataInstanceAnnotationContainerAnnotation" in the model builder? Maybe i miss something?

---
In reply to: [490435539](https://github.com/OData/WebApi/pull/2219#discussion_r490435539) [](ancestors = 490435539)

Reply:
right; we need that for the explicit model. I missed that; thanks for catching!

---
In reply to: [490438553](https://github.com/OData/WebApi/pull/2219#discussion_r490438553) [](ancestors = 490438553,490435539)